Sources/performance-worker/WorkerService.swift

```diff
@@ -98,3 +134,53 @@ final class WorkerService: Grpc_Testing_WorkerService.ServiceProtocol, Sendable
     throw RPCError(code: .unimplemented, message: "This RPC has not been implemented yet.")
   }
 }
+
+@available(macOS 13.0, iOS 16.0, watchOS 9.0, tvOS 16.0, *)
+extension WorkerService {
+  private func setupServer(_ config: Grpc_Testing_ServerConfig) async throws {
+    try self.state.withLockedValue { state in
+      switch state.role {
+      case .server(_):
+        throw RPCError(code: .alreadyExists, message: "A server has been previously set up.")
+      case .client(_):
+        throw RPCError(code: .failedPrecondition, message: "This worker has a client set up.")
+      default:
+        let server = GRPCServer(transports: [], services: [BenchmarkService()])
+        state.role = .server(server)
+      }
+    }
+    let initialStats = try await ServerStats()
```

Comment:
We have an implicit invariant at this point: when there is a server there must also be stats.

At the moment this is codified via `Role?` and `ServerStats?` which gives us many more options in code than is actually possible, e.g.:

```swift
switch (state.role, state.initialStats) {
// These are valid combinations:
case (.some(.client), .none),
  (.some(.server), .some),
  (.none, .none):
  // ...

// These are reachable but shouldn't be possible:
case (.some(.client), .some),
  (.some(.server), .none),
  (.none, .some):
  // ...
}
```

We can codify this more clearly by extending the role to include server stats, e.g.:

```swift
switch state.role {
case .some(.client):
  ()
case .some(.server, let stats):
  ()
}
```